flink-tests/src/test/java/org/apache/flink/test/scheduling/PipelinedRegionSchedulingITCase.java

```diff
@@ -204,4 +248,22 @@ public void invoke() throws Exception {
             }
         }
     }
+
+    /** Invokable which fails exactly once with a {@link PartitionException}. */
+    public static class OneTimeFailingReceiverWithPartitionException extends AbstractInvokable {
+
+        private static final AtomicBoolean hasFailed = new AtomicBoolean(false);
```

Comment:
I would suggest to make this resettable. Otherwise it won't be possible to run `testRecoverFromPartitionException` repeatedly from your IDE. Only the first run would throw the `PartitionNotFoundException` and all other runs would simply be a `NoOpInvokable`.

Reply:
You are right. I will change the case to reset this `AtomicBoolean` each time on its execution.

Reply:
done.